src/core_modules/capture-core/components/Pages/Enrollment/EnrollmentPage.container.js

```diff
@@ -4,9 +4,69 @@ import type { ComponentType } from 'react';
 import { useDispatch, useSelector } from 'react-redux';
 import { EnrollmentPageComponent } from './EnrollmentPage.component';
 import type { EnrollmentPageStatus } from './EnrollmentPage.types';
-import { cleanEnrollmentPage, fetchEnrollmentPageInformation } from './EnrollmentPage.actions';
+import {
+    cleanEnrollmentPage,
+    fetchEnrollmentPageInformation,
+    showDefaultViewOnEnrollmentPage,
+    showMissingProgramMessageOnEnrollmentPage,
+    showMissingEnrollmentMessageOnEnrollmentPage,
+    showEventProgramMessageOnEnrollmentPage,
+    showMissingCategoryMessageOnEnrollmentPage,
+    showZeroEnrollmentsMessageOnEnrollmentPage,
+} from './EnrollmentPage.actions';
+import { useScopeInfo } from '../../../hooks/useScopeInfo';
+import { scopeTypes } from '../../../metaData';
+import { useMissingCategoriesInProgramSelection } from '../../../hooks/useMissingCategoriesInProgramSelection';
+
+const useComponentLifecycle = () => {
+    const dispatch = useDispatch();
+
+    const { teiId, programId, enrollmentId } =
+      useSelector(({ router: { location: { query } } }) =>
+          ({
+              teiId: query.teidId,
+              programId: query.programId,
+              enrollmentId: query.enrollmentId,
+          }),
+      );
+
+    // todo annotate
+    const enrollments: Object = useSelector(({ enrollmentPage }) => enrollmentPage.enrollments);
+    const programHasEnrollments = enrollments && enrollments.some(({ program }) => programId === program);
+    const { scopeType } = useScopeInfo(programId);
+    const { programSelectionIsIncomplete } = useMissingCategoriesInProgramSelection();
+
+    useEffect(() => {
+        const selectedProgramIsTracker = programId && scopeType === scopeTypes.TRACKER_PROGRAM;
+        const selectedProgramIsEvent = programId && scopeType === scopeTypes.EVENT_PROGRAM;
+
+        if (selectedProgramIsTracker && programSelectionIsIncomplete) {
+            dispatch(showMissingCategoryMessageOnEnrollmentPage());
+        } else if (selectedProgramIsTracker && programHasEnrollments && enrollmentId) {
+            dispatch(showDefaultViewOnEnrollmentPage());
+        } else if (selectedProgramIsTracker && programHasEnrollments && !enrollmentId) {
+            dispatch(showMissingEnrollmentMessageOnEnrollmentPage());
+        } else if (selectedProgramIsTracker && !programHasEnrollments) {
+            dispatch(showZeroEnrollmentsMessageOnEnrollmentPage());
+        } else if (selectedProgramIsEvent) {
+            dispatch(showEventProgramMessageOnEnrollmentPage());
+        } else {
+            dispatch(showMissingProgramMessageOnEnrollmentPage());
+        }
```

Comment:
Not huge fan of this. I am happy to discuss if you have any better ideas. 
I am not sure how this can be simpler. At the moment we are having many different cases for the enrollment page and the possible messages that the user can view.

Reply:
We discussed on slack. Follow up in new PR: Create a new component to deal with showing the missing message (and keep the missing statuses locale instead of using the redux state)